source/plugin/commands/argumentparsing.rst

```diff
@@ -0,0 +1,177 @@
+================
+Argument Parsing
+================
+
+.. javadoc-import::
+    org.spongepowered.api.command.parameter.CommandContext
+    org.spongepowered.api.command.parameter.Parameter
+    org.spongepowered.api.command.parameter.Parameter.Builder
+    org.spongepowered.api.command.Command.Builder
+    java.lang.Class
+
+Minecraft's Brigadier system includes a powerful argument parser that Sponge can take advantage of. 
+It converts the string input to java base types (integer, booleans, string) or game objects 
+(players, worlds, block types, ...). Sponge's system also supports optional arguments and flags. It also handles 
+suggestions of arguments.
+
+The parsed arguments are stored in the :javadoc:`CommandContext` object. If the parser returns a single object, 
+obtain it with :javadoc:`CommandContext#one(ResourceKey)`. Optional and weak arguments may return ``Optional.empty()``
+
+Many of the parsers may return more than one object; for example, multiple players with a matching username. In that 
+case, you must use the :javadoc:`CommandContext#all(ResourceKey)` method to get the ``Collection`` of possible matches. 
+*Otherwise, the context object will throw an exception!*
+
+When creating a command, consider whether the argument could return multiple values, for example whether a player 
+argument could support multiple players when using a selector. If you support multiple values, the users need to type 
+only one command, and can use an easier command sytax, e.g ``/tell @a Who took the cookies?``
+
+To create a new :javadoc:`Parameter` (argument), use the :javadoc:`Parameter` class that will give you many 
+:javadoc:`Parameter.Builder` options. Each parameter will need its :javadoc:`Parameter.Builder#key(String)` 
+filled out before being built. 
+
+Apply the ``Parameter`` to the command builder with the :javadoc:`Command.Builder#addParameter(Parameter)` method. 
+It is possible to pass more than one ``Parameter`` to the method, thus chaining multiple arguments. 
+Example ``/msg <player> <msg>``. This has the same effect as wrapping the ``Parameter`` objects in a 
+:javadoc:`Parameter#seq(Parameter, Parameter, Parameter...)` element.
```

Comment:
This link seems to not work at all (javadoc-plugin bug)

Reply:
Changed to same method with different parameters